### DISCUSSION TITLE
How is the blade torsional degree of freedom activated? Can it be simulated just with OpenFAST or is ADAMS required for that?

### DISCUSSION THREAD

**Discussion by GBA217** [OpenFAST/openfast#2093] [Q&A]
I'm relatively new to OpenFAST and am trying to build a simple onshore 2MW model using some reference models. I would like to know whether the blade torsional DoF is available in OpenFAST or not. If yes, how do I activate it? Also can it be simulated without ADAMS? What are the channels which can be used to check whether the torisonal DoF is active or not? TIA!

**Comment by jjonkman** [comment]
Dear @gorby217,

OpenFAST supports blade torsion DOFs when enabling the BeamDyn module.  If you are modeling the blade structural dynamics in ElastoDyn, blade torsion is not a DOF.

FYI: The FAST-to-ADAMS preprocessor and ADAMS-to-AeroDyn interface is not supported by FAST v8 or OpenFAST.  We stopped supporting that after FAST v7.

Best regards,

**Reply by GBA217** [reply]
Thanks for the quick response @jjonkman! Sorry to trouble you with some more follow up questions just to get a clear picture.

I use open FAST, I chose the option to use BeamDyn for Blade, I see the blade definition including the cross sectional stiffness matrices. I understand that we can modify the entries, and this includes a torsional stiffness and, in turn, the blade has a torsional DOF. Correct up to here?Now I want to verify in our simulation that this is actually working. Can I output a blade torsional deflection angle as an output channel? For example at the blade tip? What is the respective output sensor?

**Reply by jjonkman** [reply]
Dear @gorby217,

Yes, your understanding is correct.

BeamDyn outputs rotations at structural analysis nodes (including the blade tip) as Wiener-Milenković rotational parameters.  For example, BeamDyn output `TipRDzr` provides the z component of the Wiener-Milenković rotational parameter in the "r" coordinate system (which is equivalent with the IEC blade coordinate system).

You can convert Wiener-Milenković rotational parameters to other forms, such as direction cosine angles.  See the BeamDyn documentation on Wiener-Milenković rotational parameters for more information: https://openfast.readthedocs.io/en/main/source/user/beamdyn/theory.html#wiener-milenkovic-rotation-parameter.

Regarding the torsion angle specifically, see the related discussion on our forum: https://forums.nrel.gov/t/extract-blade-tip-torsional-deformation-from-the-wiener-milenkovic-parameters-in-beamdyn/2101.

Best regards,
